locksmith/src/controllers/v2/eventCollectionController.ts

```diff
@@ -0,0 +1,211 @@
+import { RequestHandler } from 'express'
+import { z } from 'zod'
+import { EventCollection } from '../../models/EventCollection'
+import { createSlug } from '../../utils/createSlug'
+import '../../models/associations'
+import { EventData } from '../../models'
+import {
+  addEventToCollectionOperation,
+  getEventsInCollectionOperation,
+} from '../../operations/eventCollectionOperations'
+
+// schema for the event collection body
+const EventCollectionBody = z.object({
+  title: z.string(),
+  description: z.string(),
+  banner: z.string().optional(),
+  links: z
+    .array(
+      z.object({
+        name: z.string(),
+        url: z.string(),
+      })
+    )
+    .optional(),
+  managerAddresses: z.array(z.string()),
+})
+
+// schema for adding an event to a collection
+const AddEventToCollectionBody = z.object({
+  eventSlug: z.string(),
+})
+
+/**
+ * Creates a new event collection.
+ * @param req - The request object containing the event collection data.
+ * @param res - The response object used to send the response.
+ */
+export const createEventCollection: RequestHandler = async (req, res) => {
+  // Parse and validate the request body against the defined schema
+  const parsedBody = await EventCollectionBody.parseAsync(req.body)
+
+  // Generate a unique slug for the event collection based on the title
```

Comment:
I would move that logic into a `createEventCollection` operation and test it extensively there...

Reply:
definitely!